server/aesdsocket.c

```diff
@@ -0,0 +1,352 @@
+
+        //Allocate for the packet buffer
+        packet_buffer = (char *) malloc(BUFFER_SIZE*20);
+
+        while (1) {
+            bytes_received = recv(client_socket, buffer, BUFFER_SIZE, 0);
+            end_of_packet = false;
+            if (bytes_received == -1) {
+                perror("recv");
+                close(client_socket);
+                break;
+            } else if (bytes_received == 0) {
+                // Connection closed by the client
+                syslog(LOG_INFO, "Closed connection from %s \r\n", s);
+                printf("Connection closed \r\n");
+                close(client_socket);
+                break;
+            }
+            //Iterate through the bytes received and add them to the buffer to write to the file
+            for (int i = 0; i < bytes_received; i++) {
+                packet_buffer[packet_buffer_pos++] = buffer[i];
+                //Keep appending bytes till a newline is seen
+                if (buffer[i] == '\n') {
+                    packet_buffer[packet_buffer_pos] = '\0';
+                    packet_buffer_pos = 0;
+                    end_of_packet = true;
+                    break;
+                }
+
+            }
+
+            if (end_of_packet) {
+                //Echo the packet serverside just to make sure
+                printf("Incoming packet contained %s", packet_buffer);
+
+                //Create the output file if it doesn't exist
+                int log_fd = open(OUTPUT_FILE, O_WRONLY | O_CREAT | O_APPEND, 0666);
+
+                //Append the contents of the packet to the file
+                ssize_t bytes_written = write(log_fd, packet_buffer, strlen(packet_buffer));
+                if (bytes_written == -1) {
+                    perror("child process failed to write to file");
+					free(packet_buffer);
+                    close(log_fd);
+                    return 1; // Return an error code
+                }
+                close(log_fd);
+                // Check if a packet has finish being received and then send back the contents of the file
+                if (!fork()) {
+                    FILE *file;
+                    char *tmp_buffer;
+                    size_t file_size;
+
+                    //Child Process doesn't need server socket
+                    close(socket_fd);
+
+                    file = fopen(OUTPUT_FILE, "rb");
+                    if (file == NULL) {
+                        perror("writer process unable to open file");
+                        exit(EXIT_FAILURE);
+                    }
+
+
+                    // Get the file size
+                    fseek(file, 0, SEEK_END);
+                    file_size = ftell(file);
+                    rewind(file);// Move the pointer back to the start
+                    //Malloc memory to read the current file contents
+
+                    // Allocate memory for the file contents
+                    tmp_buffer = (char *) malloc(file_size);
```

Comment:
@Silex93 what happens when GB of data have been received across multiple packets?  Will this implementation meet the assignment requirements?